src/Libraries/Microsoft.Extensions.AI.Evaluation.Quality/ChatResponseExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Evaluation.Quality;
+
+internal static class ChatResponseExtensions
+{
+    internal static string RenderAsJson(this ChatResponse modelResponse, JsonSerializerOptions? options = null)
+    {
+        _ = Throw.IfNull(modelResponse);
+
+        return modelResponse.Messages.RenderAsJson(options);
+    }
+
+    internal static string RenderToolCallsAndResultsAsJson(
+        this ChatResponse modelResponse,
+        JsonSerializerOptions? options = null)
+    {
+        _ = Throw.IfNull(modelResponse);
+
+        var toolCallsAndResultsJsonArray = new JsonArray();
+
+        foreach (AIContent content in modelResponse.Messages.SelectMany(m => m.Contents))
+        {
+            if (content is FunctionCallContent or FunctionResultContent)
+            {
+                Type contentType =
+                    content is FunctionCallContent ? typeof(FunctionCallContent) : typeof(FunctionResultContent);
+
+                string toolCallOrResultJsonString =
+                    JsonSerializer.Serialize(
+                        content,
+                        AIJsonUtilities.DefaultOptions.GetTypeInfo(contentType));
+
+                JsonNode? toolCallOrResultJsonNode = JsonNode.Parse(toolCallOrResultJsonString);
+                if (toolCallOrResultJsonNode is not null)
+                {
+                    toolCallsAndResultsJsonArray.Add(toolCallOrResultJsonNode);
+                }
```

Comment:
What is the purpose of re-parsing the serialized JSON?

Reply:
Perhaps there is a better way to do this that I missed - but I am adding the serialized JSON to a `JsonArray`, and I could not figure out a way to do this without parsing it into a `JsonNode`.

Reply:
I guess that's the only route to get from a .NET object to a JsonNode?